### PR TITLE
[Merged by Bors] - feat: use junk value in `Kernel.map`, new class `IsZeroOrMarkovKernel`

### PR DESCRIPTION
**Pull request by sgouezel**
Followup to #16130, doing this at the level of kernels instead of measures.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by RemyDegenne** [comment]
It looks like this PR contains some changes that are also in #16399 (in Independence/Kernel). Please merge master once that other one is merged.

**Comment by sgouezel** [comment]
I have merged master.

**Comment by RemyDegenne** [comment]
Thanks! The new map looks much easier to use.
bors r+

File: Mathlib/Probability/Kernel/Composition.lean [hunk 1]
```diff
@@ -566,62 +567,95 @@ section MapComap
 /-! ### map, comap -/
 
 
-variable {γ δ : Type*} {mγ : MeasurableSpace γ} {mδ : MeasurableSpace δ} {f : β → γ} {g : γ → α}
+variable {γ δ : Type*} [MeasurableSpace γ] {mδ : MeasurableSpace δ} {f : β → γ} {g : γ → α}
```
Comment (YaelDillies) [inline Mathlib/Probability/Kernel/Composition.lean:570]: FYI this change broke [the GibbsMeasure project](https://github.com/james18lpc/GibbsMeasure)

Comment (YaelDillies) [reply Mathlib/Probability/Kernel/Composition.lean:570]: Reverted in #17299

**Pull request closed without merging**